### PR TITLE
Deprecate `machete.overrideForkPoint.<branch>.whileDescendantOf` key

### PR DESCRIPTION
**Pull request by PawelLipski**
See https://github.com/VirtusLab/git-machete/pull/828

### CODE REVIEW

File: backend/impl/src/test/resources/version.txt [hunk 1]
```diff
@@ -1 +1 @@
-git-machete version 3.15.1
+git-machete version 3.16.0
```
Comment (PawelLipski) [inline backend/impl/src/test/resources/version.txt:1]: Used the version from https://github.com/VirtusLab/git-machete/pull/828/ (yet unreleased)

File: testCommon/src/testFixtures/resources/setup-for-overridden-fork-point.sh [hunk 2]
```diff
@@ -25,7 +25,6 @@ cd machete-sandbox
   git branch -d root
 
   git config machete.overrideForkPoint.build-chain.to "$(git rev-parse allow-ownership-link)"
-  git config machete.overrideForkPoint.build-chain.whileDescendantOf "$(git rev-parse build-chain)"
```
Comment (PawelLipski) [inline testCommon/src/testFixtures/resources/setup-for-overridden-fork-point.sh:28]: CLI 3.15.x would have a different output for this script (since it still requires `whileDescendantOf` to be present for override to apply), hence a regeneration was needed to keep stuff consistent

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/OverrideForkPointBackgroundable.java [hunk 3]
```diff
@@ -88,6 +87,11 @@ private void setOverrideForkPointConfigValues(
     }
 
     try {
+      // As a step towards deprecation of `whileDescendantOf` key (see #1580),
```
Comment (PawelLipski) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/OverrideForkPointBackgroundable.java:90]: This was already in place for some time, actually (piggybacked with some unrelated changes).

**Review by mkondratek — APPROVED** [review]
looks good
